### PR TITLE
[store pickup] fixed some code style

### PR DESCRIPTION
**Pull request by TBlindaruk**
- add `FIXME` to app/code/Magento/InventoryInStorePickup/Model/PickupLocation/Mapper/PreProcessor/FrontendName.php, since it could cause error
 - done some refactoring of code
 - delete unnneded dependency in the app/code/Magento/InventoryInStorePickupShipping/Model/Carrier/Validation/SalesChannelValidator.php
 - add return type

### CODE REVIEW

**Comment by novikor** [comment]
@magento run static tests

**Comment by ishakhsuvarov** [comment]
@novikor It can't do that yet :) Have to fix all technical debt from the past before doing that

**Comment by novikor** [comment]
> @novikor It can't do that yet :)

Okay (

**Comment by TBlindaruk** [comment]
@ishakhsuvarov  what is wrong with functional tests?)

**Comment by ishakhsuvarov** [comment]
@TBlindaruk Unfortunately Magento's testing infrastructure was not ready to work with the branch which does not have a clear release line indication (test setups may be different for 2.1, 2.2, 2.3). It's being fixed now. Until it's fixed we would have to continue rely on Travis CI for store-pickup.

File: app/code/Magento/InventoryInStorePickupShippingApi/Model/Carrier/InStorePickup.php [hunk 1]
```diff
@@ -153,7 +153,7 @@ private function createResultMethod(float $shippingPrice): Method
     /**
      * @inheritdoc
      */
-    public function getAllowedMethods()
+    public function getAllowedMethods(): array
```
Comment (ishakhsuvarov) [inline app/code/Magento/InventoryInStorePickupShippingApi/Model/Carrier/InStorePickup.php]: This method implements `\Magento\Shipping\Model\Carrier\CarrierInterface::getAllowedMethods` which does not have `array` type. It's best to remain consistent with interfaces.

File: app/code/Magento/InventoryInStorePickup/Model/PickupLocation/Mapper/PreProcessor/FrontendName.php [hunk 2]
```diff
@@ -25,6 +25,7 @@ class FrontendName implements PreProcessorInterface
     public function process(SourceInterface $source, $value): string
     {
         if (empty($value)) {
+            // TODO: something wrong, since by the hint 'getName' could be null, but method return only string
```
Comment (ishakhsuvarov) [inline app/code/Magento/InventoryInStorePickup/Model/PickupLocation/Mapper/PreProcessor/FrontendName.php]: I'd suggest to file a ticket with description. TODOs in code rarely get monitored, tracked and fixed.

**Review by ishakhsuvarov — CHANGES_REQUESTED** [review]

File: app/code/Magento/InventoryInStorePickup/Model/PickupLocation/Mapper/PreProcessor/FrontendName.php (hunk 2 above)
Comment (TBlindaruk) [reply app/code/Magento/InventoryInStorePickup/Model/PickupLocation/Mapper/PreProcessor/FrontendName.php]: https://github.com/magento/inventory/issues/2348

**Review by novikor — APPROVED** [review]

**Pull request merged**